modules/test/playwright/fixtures/login.fixture.ts

```diff
@@ -0,0 +1,107 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {Cookie, Page, expect, test} from '@playwright/test';
+
+import {liferayConfig} from '../liferay.config';
+import {createTempFile, readTempFile} from '../utils/util';
+
+export interface Login {
+	password: string;
+	sessionId: string;
+	user: string;
+}
+
+const TIMEOUT_OPTION = {
+	timeout: 5 * 1000,
+};
+
+let loggedIn = false;
+
+const loginTest = test.extend<{
+	_login: Login;
+}>({
+	_login: async ({page}, use) => {
+		const user = liferayConfig.user.login;
+		const password = liferayConfig.user.password;
+
+		if (!loggedIn) {
+			const storageStatePath = createTempFile('storageState.json');
+
+			await page.goto('/');
+
+			await page.getByRole('button', {name: 'Sign In'}).click();
+
+			await page.getByLabel('Email Address').fill(user);
+			await page.getByLabel('Password').fill(password);
+			await page.getByLabel('Remember Me').check();
+
+			await page
+				.getByLabel('Sign In- Loading')
+				.getByRole('button', {name: 'Sign In'})
+				.click();
+
+			await maybeSetRecoveryAnswer(page);
+
+			await maybeAcceptTermsOfUse(page);
+
+			await expect(
+				page.getByLabel('Open Applications MenuCtrl+')
+			).toBeVisible(TIMEOUT_OPTION);
+
+			await page.context().storageState({path: storageStatePath});
+
+			loggedIn = true;
+		}
+		else {
+			const {cookies} = JSON.parse(readTempFile('storageState.json'));
+
+			page.context().addCookies(cookies);
+		}
+
+		const cookies = await page.context().cookies();
+
+		await use({
+			user,
+			password,
+			sessionId: cookies.find((cookie) => cookie.name === 'JSESSIONID')
+				.value,
+		});
+	},
+});
+
+async function maybeAcceptTermsOfUse(page: Page) {
+	try {
+		await expect(page.getByText('Terms of Use')).toBeVisible(
+			TIMEOUT_OPTION
+		);
+
+		await page.getByText('Done').click();
+	}
+	catch {
+
+		// Do nothing
```

Comment:
Can we emit something to the console here in case of failure(warn or log)? Just thinking that we should be overly verbose in avoiding silent failures.

Reply:
I thought about it but in this case the timeout means we don't need to do anything. I would have preferred to do it with an `if` but I couldn't find any method in playwright to do it, so I had to control the flow with `try/catch`.

In any case, if this swallows any illegitimate error, the test will fail in the next step and the screenshots will show the problem. I guess that's better than flooding the logs with errors that are not really errors 99% of the times :shrug: 

But yeah, it's a bit ugly.